flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -250,6 +255,15 @@ public CoLocationConstraint getLocationConstraint() {
 		return locationConstraint;
 	}
 
+	public InputSplit getNextInputSplit(String host) {
+		final int taskId = this.getParallelSubtaskIndex();
+		synchronized (this.inputSplits) {
+			final InputSplit nextInputSplit = this.jobVertex.getSplitAssigner().getNextInputSplit(host, taskId);
+			this.inputSplits.add(nextInputSplit);
```

Comment:
I would suggest to check for `null` here and only call add for non-null splits.

Reply:
Done.